learn/guides/calling-java-code-from-ballerina.md

```diff
@@ -0,0 +1,290 @@
+            Yaml yaml = new Yaml();
+            Map<String, Object> obj = yaml.load(inputStream);
+            System.out.println(obj);
+        } catch (Exception e) {
+            System.err.println("The file '" + filename + "' cannot be loaded. Reason: " + e.getMessage());;
+        }
+    }
+}
+```
+
+#### Creating the `FileInputStream`
+Our goal here is to create a new `java.io.FileInputStream` instance from the filename. In Step 3, we generated bindings for the required Java classes. The following is the code snippet that does the job.
+
+```ballerina
+javaio:FileInputStream | javaio:FileNotFoundException fileInputStream = javaio:newFileInputStream3(filename);
+```
+
+Here, `FileInputStream` is the Ballerina class generated for the `java.io.FileInputStream` class.
+- Ballerina bindings for each Java package are mapped onto a separate Ballerina module by default. Therefore, you need to import them when using them inside other modules. Here the `java.io` Ballerina module (mapping the corresponding Java package) is imported as `javaio`. However, if you wish to generate all the bindings inside a single directory you can do so by using the `[(-o|--output) <output-path>]` command option.
+- You can find functions that start with `newFileInputStream` in the generated code. Each such function creates a new `java.io.FileInputStream` instance. Ballerina does not support function overloading. Therefore, the bindgen tool generates a separate Ballerina function for each overloaded method or constructor. We will improve the function names of the generated bindings in a future release. 
+- All the public instance methods in the `java.io.FileInputStream` class are mapped to methods in the generated Ballerina class. You can refer [the `bindgen` tool](#the-bindgen-tool) section for more details on how other Java class members are mapped into Ballerina bindings.
+
+Next, we’ll handle the error using a type guard.
+```ballerina
+if fileInputStream is javaio:FileNotFoundException {
+	// The type of fileInputStream is FileNotFoundException within this block
+    io:println("The file '" + filename + "' cannot be loaded. Reason: " + fileInputStream.message());
+} else {
+	// The type of fileInputStream is FileInputStream within this block
+}
+```
+#### Creating the SnakeYAML Entry Point
+The `org.yaml.snakeyaml.Yaml` class is the entry point to the SnakeYAML API. The generated corresponding Ballerina class is `Yaml`. The `newYaml1()` function is mapped to the default constructor of the Java class. Let's import the `org.yaml.snakeyaml` Ballerina module as `snakeyaml`.
+```ballerina
+snakeyaml:Yaml yaml = snakeyaml:newYaml1();
+```
+####  Loading the YAML Document
+We'll be using the `org.yaml.snakeyaml.Yaml.load(InputStream is)` method to get a Java Map instance from the given `InputStream`. Since, the `Object` Ballerina class (the mapping of `java.lang.Object` class) resides inside the `java.lang` module, let's import it as `javalang`.
```

Comment:
```suggestion
We'll be using the `org.yaml.snakeyaml.Yaml.load(InputStream is)` method to get a Java Map instance from the given `InputStream`. Since the `Object` Ballerina class (the mapping of `java.lang.Object` class) resides inside the `java.lang` module, let's import it as `javalang`.
```

Reply:
Done.